src/heroku-keepalive.coffee

```diff
@@ -38,16 +44,36 @@ module.exports = (robot) ->
   if keepaliveInterval > 0.0
     robot.herokuKeepaliveIntervalId = setInterval =>
       robot.logger.info 'keepalive ping'
-      robot.http("#{keepaliveUrl}heroku/keepalive").post() (err, res, body) =>
-        if err?
-          robot.logger.info "keepalive pong: #{err}"
-          robot.emit 'error', err
+
+      if process.env.HUBOT_HEROKU_WAKEUP_TIME && process.env.HUBOT_HEROKU_SLEEP_TIME
+        wakeUpTime = process.env.HUBOT_HEROKU_WAKEUP_TIME.split(':')
+        sleepTime  = process.env.HUBOT_HEROKU_SLEEP_TIME.split(':')
+
+        wakeUpOffset = 60 * wakeUpTime[0]  + 1 * wakeUpTime[1]
+        sleepOffset  = 60 * sleepTime[0]   + 1 * sleepTime[1]
+
+        now = new Date()
+        nowOffset    = 60 * now.getHours() + now.getMinutes()
+
+        if (nowOffset >= wakeUpOffset && nowOffset < sleepOffset)
+          callKeepAlive()
         else
-          robot.logger.info "keepalive pong: #{res.statusCode} #{body}"
+          robot.logger.info "Skipping keep alive, time to rest"
+      else
+        callKeepAlive()
```

Comment:
Makes sense.

Now, if we want to have defaults for the times, then what do we want to check to see if we should even run it or not? Should we base it on HUBOT_HEROKU_KEEPALIVE_URL, where if it is available it runs, if not it doesn't? That way, in your example above, you could set that var for staging, but not for production and it should work as planned.